### DISCUSSION TITLE
Ability to have a `-SNAPSHOT` suffix

### DISCUSSION THREAD

**Discussion by ckipp01** [lefou/mill-vcs-version#62] [Ideas]
Yet another request 😆 . I'd like the ability to be able to have `-SNAPSHOT` appended to my snapshot versions. The reason for this is that if you're publishing to sonatype snapshots they require your version to actually end in `-SNAPSHOT`. You can also see this in [sbt-dynver](https://github.com/sbt/sbt-dynver#publishing-to-sonatypes-snapshots-repository-aka-sonatype-mode), which is what I'm using to using. In their readme you can see:

> If you're publishing to Sonatype sonashots then enable ThisBuild / dynverSonatypeSnapshots := true to append "-SNAPSHOT" to the version if isSnapshot is true (which it is unless building on a tag with no local changes). This opt-in exists because the Sonatype's snapshots repository requires all versions to end with -SNAPSHOT.

Do you think this could be added? Maybe inside of `format` we can add a `appendSnapshot: Boolean = false` so that users could just turn this on, and if it's detected that it's a "snapshot" then everything else stays the same but `-SNAPSHOT` gets added at the end. wdyt?

**Comment by lefou** [comment]
TBH, I though we already support `-SNAPSHOT` formatting, but I again tested it and found there is a small piece missing, as the `commitCount` can't be disabled completely.

So, the following `format`-call does almost properly render a SNAPSHOT version, except we still print the commit count.

```scala
.format(
  countSep = "-SNAPSHOT", 
  dirtyHashDigits = 0, 
  dirtySep = "", 
  revHashDigits = 0, 
  revSep = ""
)
// => "5.3.7-SNAPSHOT30
```

I tend to re-use the `commitCountPad` settings with a negative value to signal, that we don't want a commit count at all. WDYT?

```diff
-* `commitCountPad: Byte = 0` - if greater than zero, the commit count will be padded to the given length
+* `commitCountPad: Byte = 0` - if greater than zero, the commit count will be padded to the given length. If negative, no commit count will be printed 
```

```scala
.format(
  commitCountPad = -1, 
  countSep = "-SNAPSHOT", 
  dirtyHashDigits = 0, 
  dirtySep = "", 
  revHashDigits = 0, 
  revSep = ""
) 
// => 5.3.7-SNAPSHOT
```

**Reply by lefou** [reply]
We could add a `def formatMavenSnapshot` with this setup.

**Reply by lefou** [reply]
* See #65

**Reply by lefou** [reply]
* Next attempt: #66
